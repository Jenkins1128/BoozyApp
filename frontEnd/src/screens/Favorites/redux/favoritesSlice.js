import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';
const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get(`${baseUrl}/get`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
