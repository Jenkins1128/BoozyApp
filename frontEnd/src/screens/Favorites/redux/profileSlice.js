import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: []
	}
};

export const getFavoritesAsync = createAsyncThunk('getFavoritesAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.get('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/get');
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getFavoritesSlice = createSlice({
	name: 'getFavoritesAsync',
	initialState,
	reducers: {
		resetGetFavoritesRequestStatus: (state) => {
			state.currentState.getFavoritesRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getFavoritesAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				getFavoritesRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getFavoritesAsync.rejected, (state) => {
			state.currentState.getFavoritesRequestStatus = 'rejected';
		});
	}
});

export const { resetGetFavoritesRequestStatus } = getFavoritesSlice.actions;
export const selectFavoritesState = (state) => state.profile.currentState;
export default getFavoritesSlice.reducer;
