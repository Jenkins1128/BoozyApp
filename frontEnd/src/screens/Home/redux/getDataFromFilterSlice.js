import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {}
};

export const getDataFromFilterAsync = createAsyncThunk('getDataFromFilterAsync/status', async (data, { rejectWithValue }) => {
	const { location, priceType, cuisine } = data;
	try {
		const response = await axios.get(`${baseUrl}/yelp/${location}/${priceType}/${cuisine}`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getDataFromFilterSlice = createSlice({
	name: 'getDataFromFilterAsync',
	initialState,
	reducers: {
		resetDataFromFilterRequestStatus: (state) => {
			state.currentState.dataFromFilterRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getDataFromFilterAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				dataFromFilterRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getDataFromFilterAsync.rejected, (state) => {
			state.currentState.dataFromFilterRequestStatus = 'rejected';
		});
	}
});

export const { resetDataFromFilterRequestStatus } = getDataFromFilterSlice.actions;
export const selectDataFromFilterState = (state) => state.dataFromFilter.currentState;
export default getDataFromFilterSlice.reducer;
