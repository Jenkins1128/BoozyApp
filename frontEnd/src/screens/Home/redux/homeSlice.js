import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {
		location: '',
		cuisine: null,
		priceType: 6,
		$color: 'white',
		$$color: 'white',
		$$$color: 'white',
		$$$$color: 'white',
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: [],
		initialPosition: {
			latitude: 0,
			longitude: 0
		},
		markerPosition: {
			latitude: 0,
			longitude: 0
		}
	}
};

export const getCurrentLocationDataAsync = createAsyncThunk('getCurrentLocationDataAsync/status', async (data, { rejectWithValue }) => {
	const { latitude, longitude } = data;
	try {
		const response = await axios.get(`${baseUrl}/yelp/${latitude}/${longitude}`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const homeSlice = createSlice({
	name: 'getCurrentLocationDataAsync',
	initialState,
	reducers: {
		updateLocation: (state, { payload }) => {
			state.currentState.location = payload.location;
		},
		updateCuisine: (state, { payload }) => {
			state.currentState.cuisine = payload.cuisine;
		},
		updatePriceType: (state, { payload }) => {
			state.currentState.priceType = payload.priceType;
		},
		update$color: (state, { payload }) => {
			state.currentState.$color = payload.$color;
		},
		update$$color: (state, { payload }) => {
			state.currentState.$$color = payload.$$color;
		},
		update$$$color: (state, { payload }) => {
			state.currentState.$$$color = payload.$$$color;
		},
		update$$$$color: (state, { payload }) => {
			state.currentState.$$$$color = payload.$$$$color;
		},
		updateAllCategories: (state, { payload }) => {
			state.currentState.allCategories = payload.allCategories;
		},
		updateAlreadyFavorited: (state, { payload }) => {
			state.currentState.alreadyFavorited = payload.alreadyFavorited;
		},
		updateAlreadyRated: (state, { payload }) => {
			state.currentState.alreadyRated = payload.alreadyRated;
		},
		updateRestaurantsArray: (state, { payload }) => {
			state.currentState.restaurantsArray = payload.restaurantsArray;
		},
		updateInitialPosition: (state, { payload }) => {
			state.currentState.initialPosition = payload.initialPosition;
		},
		updateMarkerPosition: (state, { payload }) => {
			state.currentState.markerPosition = payload.markerPosition;
		},
		reset: (state) => {
			Object.assign(state.currentState, {
				location: '',
				cuisine: null,
				priceType: 6,
				$color: 'white',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'white'
			});
		},
		setPrice1: (state) => {
			Object.assign(state.currentState, {
				priceType: 1,
				$color: 'yellow',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'white'
			});
		},
		setPrice2: (state) => {
			Object.assign(state.currentState, {
				priceType: 2,
				$color: 'white',
				$$color: 'yellow',
				$$$color: 'white',
				$$$$color: 'white'
			});
		},
		setPrice3: (state) => {
			Object.assign(state.currentState, {
				priceType: 3,
				$color: 'white',
				$$color: 'white',
				$$$color: 'yellow',
				$$$$color: 'white'
			});
		},
		setPrice4: (state) => {
			Object.assign(state.currentState, {
				priceType: 4,
				$color: 'white',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'yellow'
			});
		},
		resetHomeRequestStatus: (state) => {
			state.currentState.locationDataRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(getCurrentLocationDataAsync.fulfilled, (state, { payload }) => {
				Object.assign(state.currentState, {
					locationDataRequestStatus: 'fulfilled',
					restaurantsArray: payload
				});
			})
			.addCase(getCurrentLocationDataAsync.rejected, (state) => {
				state.currentState.locationDataRequestStatus = 'rejected';
			});
	}
});

export const {
	updateLocation,
	updateCuisine,
	updatePriceType,
	update$color,
	update$$color,
	update$$$color,
	update$$$$color,
	updateAllCategories,
	updateAlreadyFavorited,
	updateAlreadyRated,
	updateRestaurantsArray,
	updateInitialPosition,
	updateMarkerPosition,
	reset,
	setPrice1,
	setPrice2,
	setPrice3,
	setPrice4,
	resetHomeRequestStatus
} = homeSlice.actions;

export const selectHomeState = (state) => state.home.currentState;
export default homeSlice.reducer;
