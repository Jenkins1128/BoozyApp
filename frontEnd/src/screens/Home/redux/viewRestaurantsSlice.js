import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const viewRestaurantsAsync = createAsyncThunk('viewRestaurantsAsync/status', async (data, { rejectWithValue }) => {
	const { index, state } = data;
	state.index = index;
	const restaurantId = state.restaurantsArray[index]['id'];
	const name = state.restaurantsArray[index]['name'];
	try {
		const alreadyFavoritedResponse = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/favorites', {
			restaurantId: restaurantId,
			name: name
		});
		state.alreadyFavorited = alreadyFavoritedResponse.data[0]['contains'];
		const alreadyRatedResponse = await axios.get(`https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/${state.restaurantsArray[index]['id']}/review`);
		state.alreadyRated = alreadyRatedResponse.data[0]['contains'];
		state.allCategories = '';
		for (let i = 0; i < state.restaurantsArray[index]['categories'].length; i++) {
			if (i == 0) {
				state.allCategories += state.restaurantsArray[index]['categories'][i]['title'];
			} else {
				state.allCategories += ', ' + state.restaurantsArray[index]['categories'][i]['title'];
			}
		}
		const newState = { ...state, restaurantsArray: [state.restaurantsArray[index]] };
		return newState;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const viewRestaurantsSlice = createSlice({
	name: 'viewRestaurantsAsync',
	initialState,
	reducers: {
		resetRestaurantRequestStatus: (state) => {
			state.currentState.restaurantRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(viewRestaurantsAsync.fulfilled, (state, { payload }) => {
				state.currentState = { ...state.currentState, restaurantInfo: payload, restaurantRequestStatus: 'fulfilled' };
			})
			.addCase(viewRestaurantsAsync.rejected, (state) => {
				state.currentState = { ...state.currentState, restaurantRequestStatus: 'rejected' };
			});
	}
});

export const { resetRestaurantRequestStatus } = viewRestaurantsSlice.actions;
export const selectRestaurantState = (state) => state.restaurant.currentState;

export default viewRestaurantsSlice.reducer;
