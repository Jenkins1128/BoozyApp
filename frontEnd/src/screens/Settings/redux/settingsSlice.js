import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {}
};

export const logoutAsync = createAsyncThunk('logoutAsync/status', async (data, { rejectWithValue }) => {
	try {
		const response = await axios.post('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/logout');
		return response.status;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const settingsSlice = createSlice({
	name: 'logoutAsync',
	initialState,
	reducers: {
		resetSettingsRequestStatus: (state) => {
			state.currentState.settingsRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(logoutAsync.fulfilled, (state) => {
			state.currentState.settingsRequestStatus = 'fulfilled';
		});
		builder.addCase(logoutAsync.rejected, (state) => {
			state.currentState.settingsRequestStatus = 'rejected';
		});
	}
});

export const { resetSettingsRequestStatus } = settingsSlice.actions;
export const selectSettingsState = (state) => state.settings.currentState;
export default settingsSlice.reducer;
