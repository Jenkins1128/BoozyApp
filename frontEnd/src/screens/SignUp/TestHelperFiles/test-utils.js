import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const INITIAL_STATE = {
	signup: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	},
	login: {
		currentState: {
			requestStatus: 'idle',
			isLoading: false,
			email: 'xxx@gmail.com',
			password: 'xxx'
		}
	}
}; // Or use value from root reducer
const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
	render(ui, {
		wrapper: ({ children }) => <Provider store={store}>{children}</Provider>,
		...options
	});

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender, store };
