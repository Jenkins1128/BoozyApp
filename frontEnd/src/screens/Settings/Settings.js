import React, { useEffect } from 'react';
import { StyleSheet, Text, View, TouchableOpacity, StatusBar, Image, SafeAreaView } from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import logo from '../../images/logo.png';
import { logoutAsync, resetSettingsRequestStatus, selectSettingsState } from './redux/settingsSlice';
import { useDispatch, useSelector } from 'react-redux';
import { setSignedIn } from '../../../appSlice';
import { getOS } from '../../helpers/os';

const Settings = () => {
	const state = useSelector(selectSettingsState);
	const dispatch = useDispatch();

	useEffect(() => {
		switch (state.settingsRequestStatus) {
			case 'idle':
				return;
			case 'rejected':
				break;
			case 'fulfilled':
				goToLogin();
				break;
			default:
				return;
		}
	}, [state]);

	const goToLogin = () => {
		setIsSignedInAsyncStorage('false');
		dispatch(setSignedIn({ signedIn: 'false' }));
		dispatch(resetSettingsRequestStatus());
	};

	const setIsSignedInAsyncStorage = async (value) => {
		try {
			await AsyncStorage.setItem('@isSignedIn', value);
		} catch (e) {}
	};

	const logout = () => {
		dispatch(logoutAsync());
	};

	return (
		<SafeAreaView testID={'container'} style={styles.container}>
			<View testID={'logoutButtonContainer'} style={styles.logoutButtonContainer}>
				<TouchableOpacity testID={'logoutButton'} onPress={logout} style={styles.logoutButton}>
					<Text testID={'logoutText'} style={styles.logoutText}>
						Log Out
					</Text>
				</TouchableOpacity>
				<Image testID={'logoImage'} source={logo} style={styles.logoImage} />
			</View>

			<View testID={'aboutContainer'} style={styles.aboutContainer}>
				<View testID={'aboutBoozyContainer'} style={styles.aboutBoozyContainer}>
					<Text testID={'aboutText'} style={styles.aboutText}>
						About Boozy
					</Text>
					<Text testID={'aboutInfo'} style={styles.aboutInfo}>
						Boozy is an app to help you find the best happy hour deals in your area.
					</Text>
				</View>
				<View>
					<Text testID={'creatorText'} style={styles.creatorText}>
						About Us
					</Text>
					<Text testID={'creatorInfo'} style={styles.creatorInfo}>
						Isaiah Jenkins, Hayden Miller, Chelsea Kaye Punzalan {'\n'}
						Sponsor: Jose Alvarado {'\n'}Professors: Doug Halperin and Luigi Lucaccini {'\n'}TA: Harrison Chase Keeling
						{'\n'}CS 490 Spring 2020
					</Text>
				</View>
			</View>
		</SafeAreaView>
	);
};

export default Settings;

const styles = StyleSheet.create({
	container: {
		flex: 1,
		paddingTop: getOS() === 'android' ? StatusBar.currentHeight : 0,
		backgroundColor: '#fff'
	},
	logoutButtonContainer: {
		flex: 1,
		justifyContent: 'flex-start',
		alignItems: 'flex-end',
		margin: 20
	},
	logoutButton: {
		width: 100,
		height: 60,
		borderRadius: 70,
		elevation: 8,
		color: '#fff',
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: '#EB8873'
	},
	logoutText: {
		color: '#fff',
		fontSize: 14,
		fontWeight: 'bold'
	},
	logoImage: {
		alignSelf: 'center',
		justifyContent: 'flex-start',
		width: '45%',
		height: '60%',
		marginTop: 20,
		opacity: 0.7
	},
	aboutContainer: {
		flex: 1,
		justifyContent: 'flex-start',
		marginHorizontal: 20
	},
	aboutBoozyContainer: {
		marginBottom: 20
	},
	aboutText: {
		color: '#EB8873',
		fontSize: 30,
		fontWeight: 'bold'
	},
	aboutInfo: {
		color: '#ccc',
		fontSize: 13
	},
	creatorText: {
		color: '#EB8873',
		fontSize: 30,
		fontWeight: 'bold'
	},
	creatorInfo: {
		color: '#ccc',
		fontSize: 13
	}
});
