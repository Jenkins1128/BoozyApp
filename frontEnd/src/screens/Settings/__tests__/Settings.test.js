import React from 'react';
import { reduxRender, cleanup } from '../../../TestHelperFiles/redux/test-utils';
import SettingsTestFuncs from '../../../TestHelperFiles/Functions/SettingsTestFuncs';
import axios from 'axios';
import { configureStore } from '@reduxjs/toolkit';
import { baseUrl } from '../../../helpers/constants';
import Settings from '../Settings';
import settingsReducer, { logoutAsync } from '../redux/settingsSlice';

describe('<Settings />', () => {
	let rendered;

	beforeEach(() => {
		rendered = reduxRender(<Settings />);
	});

	afterEach(cleanup);

	//snapshot
	it('should render correctly', () => {
		expect(rendered).toMatchSnapshot();
	});

	//content
	it('should wrap view with a flexible wrapper', () => {
		const containerViewComponent = rendered.getByTestId('container');
		const given = containerViewComponent.props.style;
		const result = {
			flex: 1,
			paddingTop: 0,
			backgroundColor: '#fff'
		};
		expect(result).toMatchObject(given);
	});

	it('should render logoutButtonContainer styles correctly', () => {
		const logoutButtonContainerComponent = rendered.getByTestId('logoutButtonContainer');
		const given = logoutButtonContainerComponent.props.style;
		const result = {
			flex: 1,
			justifyContent: 'flex-start',
			alignItems: 'flex-end',
			margin: 20
		};
		expect(result).toMatchObject(given);
	});

	it('should render logoutButton styles correctly', () => {
		const logoutButtonComponent = rendered.getByTestId('logoutButton');
		const given = logoutButtonComponent.props.style;
		const result = {
			width: 100,
			height: 60,
			borderRadius: 70,
			elevation: 8,
			color: '#fff',
			justifyContent: 'center',
			alignItems: 'center',
			backgroundColor: '#EB8873',
			opacity: 1
		};
		expect(result).toMatchObject(given);
	});

	it('should render logoutText styles correctly', () => {
		const logoutTextComponent = rendered.getByTestId('logoutText');
		const given = logoutTextComponent.props.style;
		const result = {
			color: '#fff',
			fontSize: 14,
			fontWeight: 'bold'
		};
		expect(result).toMatchObject(given);
	});

	it('should render logoImage styles correctly', () => {
		const logoImageComponent = rendered.getByTestId('logoImage');
		const given = logoImageComponent.props.style;
		const result = {
			alignSelf: 'center',
			justifyContent: 'flex-start',
			width: '45%',
			height: '60%',
			marginTop: 20,
			opacity: 0.7
		};
		expect(result).toMatchObject(given);
	});

	it('should render aboutContainer styles correctly', () => {
		const aboutContainerComponent = rendered.getByTestId('aboutContainer');
		const given = aboutContainerComponent.props.style;
		const result = {
			flex: 1,
			justifyContent: 'flex-start',
			marginHorizontal: 20
		};
		expect(result).toMatchObject(given);
	});

	it('should render aboutBoozyContainer styles correctly', () => {
		const aboutBoozyContainerComponent = rendered.getByTestId('aboutBoozyContainer');
		const given = aboutBoozyContainerComponent.props.style;
		const result = {
			marginBottom: 20
		};
		expect(result).toMatchObject(given);
	});

	it('should render aboutText styles correctly', () => {
		const aboutTextComponent = rendered.getByTestId('aboutText');
		const given = aboutTextComponent.props.style;
		const result = {
			color: '#EB8873',
			fontSize: 30,
			fontWeight: 'bold'
		};
		expect(result).toMatchObject(given);
	});

	it('should render aboutInfo styles correctly', () => {
		const aboutInfoComponent = rendered.getByTestId('aboutInfo');
		const given = aboutInfoComponent.props.style;
		const result = {
			color: '#ccc',
			fontSize: 13
		};
		expect(result).toMatchObject(given);
	});

	it('should render creatorText styles correctly', () => {
		const creatorTextComponent = rendered.getByTestId('creatorText');
		const given = creatorTextComponent.props.style;
		const result = {
			color: '#EB8873',
			fontSize: 30,
			fontWeight: 'bold'
		};
		expect(result).toMatchObject(given);
	});

	it('should render creatorInfo styles correctly', () => {
		const creatorInfoComponent = rendered.getByTestId('creatorInfo');
		const given = creatorInfoComponent.props.style;
		const result = {
			color: '#ccc',
			fontSize: 13
		};
		expect(result).toMatchObject(given);
	});

	//functions
	describe('goToLogin', () => {
		it('calls sideEffect', () => {
			jest.spyOn(SettingsTestFuncs, 'goToLogin');
			SettingsTestFuncs.goToLogin();
			expect(SettingsTestFuncs.goToLogin).toHaveBeenCalledTimes(1);
		});
	});

	describe('setIsSignedInAsyncStorage', () => {
		it('calls sideEffect', async () => {
			jest.spyOn(SettingsTestFuncs, 'setIsSignedInAsyncStorage');
			await SettingsTestFuncs.setIsSignedInAsyncStorage('true');
			expect(SettingsTestFuncs.setIsSignedInAsyncStorage).toHaveBeenCalledTimes(1);
			expect(SettingsTestFuncs.setIsSignedInAsyncStorage).toHaveBeenCalledWith('true');
		});
	});

	describe('logout', () => {
		it('calls sideEffect', () => {
			jest.spyOn(SettingsTestFuncs, 'logout');
			SettingsTestFuncs.logout();
			expect(SettingsTestFuncs.logout).toHaveBeenCalledTimes(1);
		});
	});

	//redux integration
	describe('redux', () => {
		it('should dispatch logoutAsync and state updated fulfilled', async () => {
			const postSpy = jest.spyOn(axios, 'post').mockResolvedValueOnce(204);
			const store = configureStore({
				reducer: {
					settings: settingsReducer
				}
			});
			await store.dispatch(logoutAsync());
			expect(postSpy).toBeCalledWith(`${baseUrl}/logout`);
			const state = store.getState();
			expect(state).toEqual({
				settings: {
					currentState: {
						settingsRequestStatus: 'fulfilled'
					}
				}
			});
		});
	});
});
