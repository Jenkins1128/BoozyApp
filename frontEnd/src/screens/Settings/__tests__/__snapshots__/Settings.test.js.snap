// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 0,
    }
  }
  testID="container"
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
        "justifyContent": "flex-start",
        "margin": 20,
      }
    }
    testID="logoutButtonContainer"
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EB8873",
          "borderRadius": 70,
          "color": "#fff",
          "elevation": 8,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "width": 100,
        }
      }
      testID="logoutButton"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
        testID="logoutText"
      >
        Log Out
      </Text>
    </View>
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "height": "60%",
          "justifyContent": "flex-start",
          "marginTop": 20,
          "opacity": 0.7,
          "width": "45%",
        }
      }
      testID="logoImage"
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
        "marginHorizontal": 20,
      }
    }
    testID="aboutContainer"
  >
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
      testID="aboutBoozyContainer"
    >
      <Text
        style={
          Object {
            "color": "#EB8873",
            "fontSize": 30,
            "fontWeight": "bold",
          }
        }
        testID="aboutText"
      >
        About Boozy
      </Text>
      <Text
        style={
          Object {
            "color": "#ccc",
            "fontSize": 13,
          }
        }
        testID="aboutInfo"
      >
        Boozy is an app to help you find the best happy hour deals in your area.
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "#EB8873",
            "fontSize": 30,
            "fontWeight": "bold",
          }
        }
        testID="creatorText"
      >
        About Us
      </Text>
      <Text
        style={
          Object {
            "color": "#ccc",
            "fontSize": 13,
          }
        }
        testID="creatorInfo"
      >
        Isaiah Jenkins, Hayden Miller, Chelsea Kaye Punzalan 
        

        Sponsor: Jose Alvarado 
        

        Professors: Doug Halperin and Luigi Lucaccini 
        

        TA: Harrison Chase Keeling
        

        CS 490 Spring 2020
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
